app/src/main/java/org/simple/clinic/home/overdue/removepatient/RemoveAppointmentSheet.kt

```diff
@@ -82,11 +96,18 @@ class RemoveAppointmentSheet : BottomSheetActivity() {
           .map { PatientDeadClicked(patientUuid = intent.extras!!.getSerializable(KEY_PATIENT_UUID) as UUID) }
 
   private fun cancelReasonClicks(): Observable<UiEvent> {
-    val movedOutStream = RxView.clicks(movedOutRadioButton).map { CancelReasonClicked(Moved) }
-    val notRespondingStream = RxView.clicks(notRespondingRadioButton).map { CancelReasonClicked(PatientNotResponding) }
-    val otherStream = RxView.clicks(otherReasonRadioButton).map { CancelReasonClicked(Other) }
-
-    return Observable.merge(movedOutStream, notRespondingStream, otherStream)
+    val reasonClicks = { button: RadioButton, reason: AppointmentCancelReason ->
+      RxView.clicks(button).map { CancelReasonClicked(reason) }
+    }
+
+    return Observable.mergeArray(
```

Comment:
Nice!

I took a stab at making this a bit more readable by not needing to invoke the `reasonClicks` method each time. What do you think?
```kotlin
  private fun cancelReasonClicks2(): Observable<UiEvent> {
    val buttonToCancelReason = mapOf(
        movedOutRadioButton to Moved,
        notRespondingRadioButton to PatientNotResponding,
        invalidPhoneNumberRadioButton to InvalidPhoneNumber,
        publicHospitalTransferRadioButton to TransferredToAnotherPublicHospital,
        movedToPrivateRadioButton to MovedToPrivatePractitioner,
        otherReasonRadioButton to Other)

    val reasonClicks = { entry: Map.Entry<View, AppointmentCancelReason> ->
      RxView.clicks(entry.key).map {
        CancelReasonClicked(entry.value)
      }
    }

    return Observable.merge(buttonToCancelReason.map(reasonClicks))
}
```

Reply:
Sure, looks good. Can you please help me by making these changes to the branch and merging it in if possible?

Reply:
I'd rename `buttonToCancelReason` to `buttonToCancelReasons` (plural).

Reply:
I'll do these today if I have time. I'll let you know if I pick it up.